### DISCUSSION TITLE
Uploader allowd filetype, max filesize and file number limit help

### DISCUSSION THREAD

**Discussion by jch02140** [rsuite/rsuite#1741] [Q&A]
Hi,

I am wondering if there is a way to set the maximum file size per file and the total number of file allowed to upload?

Also, I am not sure how to set allowed file type in the uploader. Anyone able to help?

**Comment by simonguo** [comment]
You can use the `shouldUpload` and `shouldQueueUpdate` callback functions to verify  file.
https://rsuitejs.com/components/uploader#File%20check

**Reply by jch02140** [reply]
Hi,

Thank you for the quick reply. 

It seems I can also able to set a a limit to the number of files allowed for uploading with

`disabled={fileList.length < MAX_NO_OF_FILES}`

as per the post here:
https://github.com/rsuite/rsuite/issues/673

I am still a little lost as to how to restrict the file size...

**Reply by jch02140** [reply]
Just a follow up. I noticed that even when I set `accept="image/jpeg, application/pdf"` which allows only jpeg/jpg and pdf files, I can still upload other file types if I choose "All Files" option in the file dialog:

Example:
https://codesandbox.io/s/rsuite-template-forked-ledi9?file=/src/index.js

**Reply by SevenOutman** [reply]
> I can still upload other file types if I choose "All Files" option in the file dialog:

This is browser and system behavior that we have no control.
